### PR TITLE
Anchor click interceptor and navigation service cleanup

### PR DESCRIPTION
**Pull request by alabiaga**
- delete anchor-click-interceptor.js and move logic to navigation.js

See https://github.com/ampproject/amphtml/issues/13772#issuecomment-370010915

### CODE REVIEW

**Comment by alabiaga** [comment]
just went the easy route with the static method but probably adoptServiceForEmbedIfEmbeddable is the right solution and then moving the maybeExpandUrlParams in  navigation.js#handle_(e)?

**Comment by jridgewell** [comment]
I don't understand what `maybeExpandUrlParams` is trying to do differently than `Navigations`'s handlers.

**Comment by lannka** [comment]
re: maybeExpandUrlParams

The function is coupled with A4A logic. It will be nice if we can clean it out, or replace with something generic.

**Comment by alabiaga** [comment]
@lannka thanks for reviewing. There is some logic that can be shared between the handlers and I'll do that refactoring in another PR. Thanks

File: src/service/navigation.js [hunk 1]
```diff
@@ -120,6 +127,17 @@ export class Navigation {
...
+  /**
+   * Registers a handler that performs URL replacement on the href
+   * of an ad click.
+   * @param {!./ampdoc-impl.AmpDoc} ampdoc
+   * @param {!Window} win
+   */
+  static installAnchorClickInterceptor(ampdoc, win) {
+    win.document.documentElement.addEventListener('click',
+        maybeExpandUrlParams.bind(null, ampdoc), /* capture */ true);
+  }
```
[the remark this answers is not in the document]
Comment (alabiaga) [reply src/service/navigation.js]: sure thing. Will send a PR for this tomorrow.

[the remark this answers is not in the document]
Comment (alabiaga) [reply src/service/navigation.js]: actually, this needs to exist as is. The click listener in this case is not on the main doc but in the iframe doc where the ad resides. This click listener is for the ad related extensions, for the creative in an iframe.

**Pull request merged**